src/regex/engine.cr

```diff
@@ -1,4 +1,8 @@
-{% if flag?(:use_pcre2) %}
+# The following condition ensures that the engine selection respects `-Duse_pcre2`/`-Duse_pcre`,
+# and if none is given it tries to check for availability of `libpcre2` with `pkg-config`.
+# If `pkg-config` is unavailable, the default is PCRE2. If `pkg-config` is available but
+# has no information about a `libpcre2` package, it falls back to PCRE.
+{% if flag?(:use_pcre2) || (!flag?(:use_pcre) && !flag?(:win32) && `hash pkg-config 2> /dev/null && (pkg-config --silence-errors --modversion libpcre2-8 || printf %s false) || true` != "false") %}
```

Comment:
Why doesn't Windows use PCRE2 by default? 1.7.0 should already come with a pre-built PCRE2

Reply:
Oh right. I didn't revisit the logic excluding Windows since #12847 was merged.

I think we can always use PCRE2 without checks (unless `-Duse_pcre`, of course) because we're shipping the library with the distribution package.